core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -32,6 +32,18 @@ private CatalogProperties() {}
   public static final String VIEW_DEFAULT_PREFIX = "view-default.";
   public static final String METRICS_REPORTER_IMPL = "metrics-reporter-impl";
 
+  /** http proxy configuration for rest catalog */
+  public static final String PROXY_ENABLED = "proxy-enabled";
+
+  public static final boolean PROXY_ENABLED_DEFAULT = false;
+  public static final String PROXY_HOSTNAME = "proxy.hostname";
+  public static final String PROXY_PORT = "proxy.port";
+
+  public static final String PROXY_REQUIRES_CREDENTIALS = "proxy.requires-credentials";
```

Comment:
This property could maybe be inferred from the presence (or absence) of `PROXY_USERNAME` and `PROXY_PASSWORD`.